### PR TITLE
aws batch job definition

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

This pr contains the singular and plural resource for batch job definitions.

### Issues Resolved

No issues.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/aws_batch_job_definition.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+| container_properties (instance_type) | The instance type of the container properties of the job definition. |
+| container_properties (resource_requirements) | The resource requirements of the container properties of the job definition. |
+| container_properties (linux_parameters (shared_memory_size)) | The shared_memory_size of the linux parameters of the container properties of the job definition. |
+| container_properties (linux_parameters (tmpfs)) | The tmpfs of the linux parameters of the container properties of the job definition. |
+| container_properties (linux_parameters (max_swap)) | The max_swap of the linux parameters of the container properties of the job definition. |
+| container_properties (linux_parameters (swappiness)) | The swappiness of the linux parameters of the container properties of the job definition. |
+| container_properties (log_configuration (options)) | The options of the log configuration of the container properties of the job definition. |
+| container_properties (log_configuration (shared_memory_size)) | The shared_memory_size of the log configuration of the container properties of the job definition. |
```
Comment (IanMadd) [inline docs/resources/aws_batch_job_definition.md:57]: Is shared_memory_size a log configuration or Linux parameter? It looks like Linux.

File: docs/resources/aws_batch_job_definitions.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+### Ensure a job_definition name is available.
+    describe aws_batch_job_definitions do
+      its('job_definition_names') { should include 'test1' }
+    end
+
+### Ensure that the state is `ENABLED` or `DISABLED`.
+    describe aws_batch_job_definitions do
+        its('statuses') { should include 'ACTIVE' }
+    end
```
Comment (IanMadd) [inline docs/resources/aws_batch_job_definitions.md:47]: Is this right? Should the heading be: Ensure that the status is `ACTIVE`.

File: docs/resources/aws_batch_job_definition.md (hunk 1 above)
Comment (soumyo13) [reply docs/resources/aws_batch_job_definition.md:57]: Yes, shared_memory_size is a Linux parameter.

File: docs/resources/aws_batch_job_definitions.md (hunk 2 above)
Comment (soumyo13) [reply docs/resources/aws_batch_job_definitions.md:47]: Yes.

**Review by IanMadd — APPROVED** [review]

**Review by Nirbhay1997 — APPROVED** [review]

**Pull request merged**